src/index.ts

```diff
@@ -44,11 +65,20 @@ export namespace Validators {
 }
 
 export namespace Utilities {
+
+    export function applyValidators(source: any, target: any, rules: { [key: string]: { type: string } }) {
```

Comment:
I have to export the function to use it in other namespaces, but I think the namespace could stay internal.

The problem with that generic is that the source/target might be an instance of a class which doesn't have index types...

Reply:
Hmmm trying to imagine a scenario where you're deserializing a JSON into a class type that isn't string indexable. You can't for instance get something like this:

![image](https://user-images.githubusercontent.com/31600776/95131535-d5f84700-0723-11eb-923b-354ba29bc75d.png)

From something like this:
```json
{
  1: "hello"
}
```
Because the latter isn't valid json, so I'm not sure what corner case this would be covering.

If there is no such corner case then maybe you could make the generic even more spicy

`<TKey extends string, T extends { [Key in TKey]: any } & { new () => T }>`

That might be exactly right idk